enumerables.rb

```diff
@@ -0,0 +1,137 @@
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity, Metrics/ModuleLength
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < size
+      if is_a? Array
+        yield self[i]
+      elsif is_a? Hash
+        yield keys[i], self[keys[i]]
+      elsif is_a? Range
+        yield to_a[i]
+      end
+      i += 1
+    end
+    self
+  end
```

Comment:
#my_each when a block is given when self is a hash behaves like #each calls the given block once for each element in self